dashboard/app/models/user_level.rb

```diff
@@ -130,28 +130,29 @@ def before_unsubmit
     end
   end
 
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`
+  def locked
+    unlocked_at.nil? || unlocked_at < AUTOLOCK_PERIOD.ago
+  end
+
   def locked=(val)
-    new_time = if val
-                 nil
-               else
-                 Time.now
-               end
-    self.unlocked_at = new_time
-  end
-
-  def locked?
-    if unlocked_at.nil?
-      return true
-    else
-      return unlocked_at < AUTOLOCK_PERIOD.ago
-    end
+    self.unlocked_at = val ? nil : Time.now
```

Comment:
again, i didn't change the logic here, just collapsed it to one line.

also, you'll note that i was completely wrong about not needing `self.` here!

Reply:
Haha fwiw- I had to try several iterations to see which would actually stick before I realized it needed 'self'.